### ISSUE TITLE
RuntimeWarning: Mean of empty slice

### ISSUE THREAD

**Issue by vedal** [HReynaud/UVT#2]
Hi!
Thanks for sharing the code of this project!

I'm trying to run training on the EchoNet dataset, but running into an error:
```
Train: False repeat reg
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1235/1235 [03:05<00:00,  6.66it/s, 0.0604 / 0.1006 0.0003 / 0.0436]Testing on reg repeat False
Using device: cuda
UVT contains 346871234 parameters.
 16%|█████████████████████████████████████████████████▌                                                                                                                                                                                                                                                                | 198/1224 [00:30<02:29,  6.89it/s]/
/.pyenv/versions/lib/python3.8/site-packages/numpy/core/fromnumeric.py:3372: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
/.pyenv/versions/lib/python3.8/site-packages/numpy/core/_methods.py:170: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
/.pyenv/versions/lib/python3.8/site-packages/numpy/core/_methods.py:233: RuntimeWarning: Degrees of freedom <= 0 for slice
  ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
/.pyenv/versions/lib/python3.8/site-packages/numpy/core/_methods.py:194: RuntimeWarning: invalid value encountered in true_divide
  arrmean = um.true_divide(
/.pyenv/versions/lib/python3.8/site-packages/numpy/core/_methods.py:226: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1224/1224 [03:07<00:00,  6.53it/s]Saved to ./output/UVT_R_REG/test.csv
Misses: 371
          Pred  Clean Pred
MAE   5.568126         0.0
RMSE  7.493932         0.0
R²    0.615646         0.0
std 5.01547550610934
ES Temporal error (s): nan
ED Temporal error (s): 0.481228142655858
     ED          ES
aFD NaN   26.680556
std NaN  137.778294
   Tolerance  Small   Small %  Large   Large %
0        0.0  920.0  0.751634  670.0  0.547386
1        1.0  920.0  0.751634  670.0  0.547386
2        2.0  920.0  0.751634  671.0  0.548203
3        3.0  920.0  0.751634  671.0  0.548203
4        5.0  921.0  0.752451  671.0  0.548203
5       10.0  959.0  0.783497  787.0  0.642974
Rejected: 0
```

This seemingly happens at the same place each epoch. I also always get "nan" for the ED prediction. The only thing I changed in main.py is that I put `batch_size=1`. 

Do you have some insights into what could be the reason?

**Comment by HReynaud** [comment]
Hello,
Lowering the batch size may have made the training unstable. My guess from your stack trace is that the network can't find any ES frames, producing an empty list of frames and resulting in the "Mean of empty slice" error and the NaN `ES Temporal error`. My suggestion would be to lower the learning rate, to compensate for the smaller batch size, may be `lr = 5e-6` will do the trick. Good luck and let me know if this remains an issue.

**Comment by vedal** [comment]
I managed to make it run with `batch_size=2` without OOM error by changing `intermediate_size = 8196` to `intermediate_size = 8192` for training (was this a typo? seems its 8192 for test, and 8196 isnt divisible by 2. Ref. #1  ).
However, the NaNs remain. I'm training on EchoNet-Dynamic without making any further changes, so its a bit strange...
Does it not happen on your side?

**Comment by HReynaud** [comment]
Thanks for spotting the typo, I pushed the correction and retrained a network with the exact code from the current main.py file. It trained for 5 epochs and I never saw a NaN appear.

```
Misses: 119
          Pred  Clean Pred
MAE   6.335590         0.0
RMSE  8.936288         0.0
R²    0.450909         0.0
std 6.3021850973198585
ES Temporal error (s): 0.15395873739513197
ED Temporal error (s): 0.06256903118815292
            ED        ES
aFD   7.868852  3.226230
std  30.902425  5.770868
   Tolerance   Small   Small %   Large   Large %
0        0.0    98.0  0.080328   180.0  0.147541
1        1.0   302.0  0.247541   478.0  0.391803
2        2.0   529.0  0.433607   711.0  0.582787
3        3.0   721.0  0.590984   882.0  0.722951
4        5.0   925.0  0.758197  1076.0  0.881967
5       10.0  1050.0  0.860656  1185.0  0.971311
Rejected: 4
```
Not sure where the error can come from.

**Comment by thiago092** [comment]
![image](https://user-images.githubusercontent.com/16467782/177664145-f9d8d546-0c78-4cd5-aa57-aaabef745c36.png)

I think it worked, thank you very much for your help and your work!
